### ISSUE TITLE
[0.18-rc][packager] Packager throws error "Cannot read property 'error' of null" instead of babel errors

### ISSUE THREAD

**Issue by corbt** [facebook/react-native#5191]
This is to track the error first reported in #5188 and #5190. (The thread in #5188 got long because we weren't sure if it was related to the simultaneous but unrelated babel classProperties syntax change, so I'm consolidating here.)

There's a regression in the packager in 0.18-rc where babel syntax errors aren't passed through to the user, making it difficult to fix the underlying bugs. These are the error messages for the same bug running `react-native bundle`:

## React Native 0.17

```
uncaught error Error: SyntaxError: /Users/kyle/proj/chimera/app/actions/auth.js: 'import' and 'export' may only appear at the top level (22:0)
  20 |
  21 |
> 22 | export function logOut() {
     | ^
  23 |   return {type: LOG_OUT};
  24 | }
```

## React Native 0.18-rc
```
uncaught error Error: TypeError: Cannot read property 'error' of null
    at index.js:117:20
    at tryCallOne (/Users/kyle/proj/chimera/node_modules/react-native/node_modules/promise/lib/core.js:37:12)
    at /Users/kyle/proj/chimera/node_modules/react-native/node_modules/promise/lib/core.js:123:15
    at flush (/Users/kyle/proj/chimera/node_modules/react-native/node_modules/promise/node_modules/asap/raw.js:50:29)
    at doNTCallback0 (node.js:428:9)
TransformError: /Users/kyle/proj/chimera/app/actions/auth.js: Cannot read property 'error' of null
See logs /var/folders/j4/wct0sgyx0p9803_1nkhm3pzm0000gn/T/react-packager.log
    at SocketClient._handleMessage (SocketClient.js:139:23)
    at BunserBuf.<anonymous> (SocketClient.js:53:42)
    at emitOne (events.js:77:13)
    at BunserBuf.emit (events.js:169:7)
    at BunserBuf.process (/Users/kyle/proj/chimera/node_modules/react-native/node_modules/bser/index.js:289:10)
    at /Users/kyle/proj/chimera/node_modules/react-native/node_modules/bser/index.js:244:12
    at doNTCallback0 (node.js:428:9)
    at process._tickCallback (node.js:357:13)
```

**Comment by ide** [comment]
cc @martinbigio, it looks like the packager has an error while trying to report errors.

**Comment by corbt** [comment]
For what it's worth I just tested with the same error on master and it looks like this has already been fixed, so it's probably just a question of figuring out what to cherry-pick into the 0.18 branch.

**Comment by satya164** [comment]
Also I recently noticed that packager no longer reports errors when it fails to import/require a module.

**Comment by floatinghotpot** [comment]
I also encounter the same issue on react-native 0.18.0-rc. 

It's very easy to re-produce it, just create a new project, `react-native init HelloReact`, open the ios project with xcode, just build, it will report exact the same error.

If change package.json the react-native version from 0.18.0-rc to 0.17.0, and update the node-modules/react-native by `rm -rf node-modules; npm install;`, open ios project with xcode, build again, now build pass.

So, it must be something wrong with react-native v0.18.0-rc.

**Comment by israelidanny** [comment]
I'm getting the exact same error by updating react 0.17 to 0.18.0-rc

**Comment by mkonicek** [comment]
Looks like this is something we should fix asap, I'll ping @martinbigio about it.

**Comment by mkonicek** [comment]
Thanks so much for using the rc everyone!

**Comment by tdzl2003** [comment]
Maybe it's caused by `.babelrc` from module `react-deep-force-update` which is incompatible with babel 6.x

**Comment by martinbigio** [comment]
This was already fixed on master last week. We need to cherry pick https://github.com/facebook/react-native/commit/5a4ee0bfaa6e6db7a105362a06fb903efa5f58a4 :)

**Comment by chshouyu** [comment]
thanks all, waiting for release.

**Comment by jinzhaoyu** [comment]
When i upgrade react-native from 0.17.0 to 0.18.0-rc , I faced same problem , and  i try to apply [5a4ee0b](https://github.com/facebook/react-native/commit/5a4ee0bfaa6e6db7a105362a06fb903efa5f58a4) ,I got babel error about `.babelrc.stage`   , see flow error info: 
> 
{"message":"TransformError: /Users/jinzhaoyu/Desktop/tt/node_modules/react-native/node_modules/react-transform-hmr/node_modules/react-proxy/node_modules/react-deep-force-update/lib/index.js: [BABEL] /Users/jinzhaoyu/Desktop/tt/node_modules/react-native/node_modules/react-transform-hmr/node_modules/react-proxy/node_modules/react-deep-force-update/lib/index.js: Unknown option: /Users/jinzhaoyu/Desktop/tt/node_modules/react-native/node_modules/react-transform-hmr/node_modules/react-proxy/node_modules/react-deep-force-update/.babelrc.stage","type":"TransformError","errors":[{}]}


my environment:

1. React-Native: 0.18-rc
2. node:  5.3.0
3. npm:  2.14.15
4. watchman: 4.3.0

**Comment by martinbigio** [comment]
@jinzhaoyu, the error you're getting is unrelated to this issue. The problem is that you have a `.babelrc` file on `react-deep-force-update`.Try removing that file. Also make sure you're using the latest version of that npm module as Dan Abramov recently released a new version upgrading that :)

**Comment by int64ago** [comment]
+1, waiting for fixing, thx :-)

**Comment by mylonelyplanet** [comment]
I think it's still not working in formal release v0.18,

**Comment by corbt** [comment]
0.18 is out now and this problem has been fixed!

**Comment by int64ago** [comment]
@corbt 
Hi, I tried the 0.18.0, but it wasn't fixed.

**Comment by jinzhaoyu** [comment]
The problem still exists in 0.18.0 ,when i execute 'bundle' command ,i got same as 0.18-rc error: 
> TransformError: /test/react-native/node_modules/react-native/node_modules/react-transform-hmr/node_modules/react-proxy/modules/index.js: [BABEL] /test/react-native/node_modules/react-native/node_modules/react-transform-hmr/node_modules/react-proxy/modules/index.js:  Unknown option: /test/react-native/node_modules/react-native/node_modules/react-transform-hmr/node_modules/react-proxy/.babelrc.stage

**Comment by tdzl2003** [comment]
Still need to remove .babelrc manually.

**Comment by corbt** [comment]
@int64ago that looks like a different bug. It does look like you're running into some kind of trouble though, do you run into that error on a fresh project?

@jinzhaoyu this is also a different bug, notice that the error message is different. Would you mind opening a new issue with steps to reproduce? It looks like it may be related to #5040 but that should be fixed. For now you can remove the offending /test/react-native/node_modules/react-native/node_modules/react-transform-hmr/node_modules/react-proxy/.babelrc file like @tdzl2003 said which should unblock you.

**Comment by int64ago** [comment]
It's the same bug as #5259 which referenced by this issue.
And it's a fresh project.

**Comment by corbt** [comment]
From #5259:
> if you go to http://localhost:8081/index.android.bundle?platform=android&dev=true in your browser...does it spit out JS? Or an error?

Can you do that and tell me what you see?

**Comment by int64ago** [comment]
```
{
  message: "TransformError: /Users/patscanlan/repos/AwesomeProject/index.android.js: Cannot read property 'error' of null",
  type: "TransformError",
  errors: [{ }]
}
```

**Comment by corbt** [comment]
Ok try the following from the command line and post back the response here?:

```
react-native bundle --entry-file=/Users/patscanlan/repos/AwesomeProject/index.android.js --bundle-output /dev/null
```

**Comment by int64ago** [comment]
Ok, I tried again with a new project start with `react-native init AwsomeProject`
After `npm install && npm start`, I open http://localhost:8081/index.android.bundle?platform=android&dev=true in my browser and I got this:

```
{
message: "TransformError: /home/int64ago/AwsomeProject/node_modules/react-deep-force-update/lib/index.js: [BABEL] /home/int64ago/AwsomeProject/node_modules/react-deep-force-update/lib/index.js: Unknown option: /home/int64ago/AwsomeProject/node_modules/react-deep-force-update/.babelrc.stage",
type: "TransformError",
errors: [
{ }
]
}
```

And then I run `react-native bundle --entry-file=/home/int64ago/AwsomeProject/index.android.js --bundle-output /dev/null` and I got this:

```
bundle: Created ReactPackager
uncaught error Error: ReferenceError: [BABEL] /home/int64ago/AwsomeProject/node_modules/react-deep-force-update/lib/index.js: Unknown option: /home/int64ago/AwsomeProject/node_modules/react-deep-force-update/.babelrc.stage
    at Logger.error (/home/int64ago/AwsomeProject/node_modules/babel-core/lib/transformation/file/logger.js:41:11)
    at OptionManager.mergeOptions (/home/int64ago/AwsomeProject/node_modules/babel-core/lib/transformation/file/options/option-manager.js:262:18)
    at OptionManager.addConfig (/home/int64ago/AwsomeProject/node_modules/babel-core/lib/transformation/file/options/option-manager.js:221:10)
    at OptionManager.findConfigs (/home/int64ago/AwsomeProject/node_modules/babel-core/lib/transformation/file/options/option-manager.js:364:16)
    at OptionManager.init (/home/int64ago/AwsomeProject/node_modules/babel-core/lib/transformation/file/options/option-manager.js:412:12)
    at File.initOptions (/home/int64ago/AwsomeProject/node_modules/babel-core/lib/transformation/file/index.js:191:75)
    at new File (/home/int64ago/AwsomeProject/node_modules/babel-core/lib/transformation/file/index.js:122:22)
    at Pipeline.transform (/home/int64ago/AwsomeProject/node_modules/babel-core/lib/transformation/pipeline.js:42:16)
    at transform (/home/int64ago/AwsomeProject/node_modules/react-native/packager/transformer.js:59:24)
TransformError: /home/int64ago/AwsomeProject/node_modules/react-deep-force-update/lib/index.js: [BABEL] /home/int64ago/AwsomeProject/node_modules/react-deep-force-update/lib/index.js: Unknown option: /home/int64ago/AwsomeProject/node_modules/react-deep-force-update/.babelrc.stage
See logs /tmp/react-packager.log
    at SocketClient._handleMessage (SocketClient.js:139:23)
    at BunserBuf.<anonymous> (SocketClient.js:53:42)
    at emitOne (events.js:77:13)
    at BunserBuf.emit (events.js:169:7)
    at BunserBuf.process (/home/int64ago/AwsomeProject/node_modules/bser/index.js:289:10)
    at /home/int64ago/AwsomeProject/node_modules/bser/index.js:244:12
    at doNTCallback0 (node.js:419:9)
    at process._tickCallback (node.js:348:13)

```

Here's my environment:
Ubuntu 14.04 x86_64 LTS
Node v4.2.4
NPM 3.5.3
react-native-cli: 0.1.10
react-native: 0.18.0

**Comment by int64ago** [comment]
@tdzl2003 
  
 > Still need to remove .babelrc manually.

It works.

**Comment by wdxgtsh** [comment]
I have remove .babelrc manually  . but no use.

**Comment by martinbigio** [comment]
@gaearon didn't you ship a version of `react-deep-force-update` without the `.bablerc`?

**Comment by 940825** [comment]
@int64ago rm project_folder/node_modules/react-deep-force-update/.bablerc

**Comment by citysheep** [comment]
for me it still didnt work after removing .bablerc

**Comment by ShadowZheng** [comment]
@martinbigio yeah, it still didn't work after removing .bablerc file. when I remove it and execute the `react-native start` it still have an error:
```bash
c:\Users\azheng\Desktop\reactNativeApp\node_modules\babel-core\lib\transformation\file\logger.js:41
    throw new Constructor(this._buildMessage(msg));
    ^

ReferenceError: [BABEL] c:\Users\azheng\Desktop\reactNativeApp\node_modules\react-native\local-cli\bundle\bundle.js: Unknown option: base.stage
    at Logger.error (c:\Users\azheng\Desktop\reactNativeApp\node_modules\babel-core\lib\transformation\file\logger.js:41:11)
    at OptionManager.mergeOptions (c:\Users\azheng\Desktop\reactNativeApp\node_modules\babel-core\lib\transformation\file\options\option-manager.js:262:18)
    at OptionManager.init (c:\Users\azheng\Desktop\reactNativeApp\node_modules\babel-core\lib\transformation\file\options\option-manager.js:416:10)
    at File.initOptions (c:\Users\azheng\Desktop\reactNativeApp\node_modules\babel-core\lib\transformation\file\index.js:191:75)
    at new File (c:\Users\azheng\Desktop\reactNativeApp\node_modules\babel-core\lib\transformation\file\index.js:122:22)
    at Pipeline.transform (c:\Users\azheng\Desktop\reactNativeApp\node_modules\babel-core\lib\transformation\pipeline.js:42:16)
    at Object.transformFileSync (c:\Users\azheng\Desktop\reactNativeApp\node_modules\babel-core\lib\api\node.js:116:10)
    at compile (c:\Users\azheng\Desktop\reactNativeApp\node_modules\babel-register\lib\node.js:102:20)
    at loader (c:\Users\azheng\Desktop\reactNativeApp\node_modules\babel-register\lib\node.js:130:14)
    at Object.require.extensions.(anonymous function) [as .js] (c:\Users\azheng\Desktop\reactNativeApp\node_modules\babel-register\lib\node.js:140:7)
```

my environment is :
react-native-cli: 0.1.10
react-native: 0.19.0
node: 4.0.0

**Comment by martinbigio** [comment]
@ShadowZheng does this repro after removing your node modules doing npm install? It's possible that you fetched a module that has a .babelrc. Also please do a search on the node_modules directory and make sure there's no file named .babelrc

**Comment by citysheep** [comment]
it worked after i followed this http://stackoverflow.com/questions/34975286/reactnativecli-init-helloworld-project-build-error-with-babel-transformerror
